### PR TITLE
Changed package to callback abstract

### PR DESCRIPTION
**Pull request by XedinUnknown**
Aims to incorporate solution for #2.

### CODE REVIEW

**Review by mecha — CHANGES_REQUESTED** [review]
Kindly remove `.idea/misc.xml` from version control and rename `.idea/callback-aware-trait.iml` to `.idea/callback-abstract.iml`.

Also, I've been noticing that `.idea/php.xml` does not use the `$PROJECT_DIR$` placeholder. Maybe we ought to stop committing that and adding it to `.gitignore`.

The only thing that PHPStorm needs to recognize that a folder is a project is the `.idea/*.iml` file.

**Comment by mecha** [comment]
@XedinUnknown Actually, my projects' `.idea/php.xml` files are correctly using the `$PROJECT_DIR$` placeholder. Perhaps your paths are incorrect?

**Comment by XedinUnknown** [comment]
@mecha, not sure exactly. Just removed the `misc.xml` for now.

File: src/InvokeCallbackCapableTrait.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        $callback = $this->_getCallback();
+
+        try {
+            return $this->_invokeCallable($callback, $args);
+        } catch (InvalidArgumentException $e) {
+            /* We know that `$args` is correct, so the only way
+             * `_invokeCallback()` would throw `InvalidArgumentException`
+             * is if the callback is wrong. But we cannot let it bubble
+             * up, because it is not an argument to this method. Therefore,
+             * catch it, wrap, and trow a more appropriate exception.
```
Comment (mecha) [inline src/InvokeCallbackCapableTrait.php]: Typo in _"trow"_

File: src/InvokeCallbackCapableTrait.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            );
+        }
+    }
+
+    /**
+     * Retrieves the callback.
+     *
+     * @since [*next-version*]
+     *
+     * @return callable
```
Comment (mecha) [inline src/InvokeCallbackCapableTrait.php]: Missing return description.

**Review by mecha — CHANGES_REQUESTED** [review]
Over all, I love this. I'm only requesting a couple of small things relating to docs.

**Review by mecha — APPROVED** [review]

**Pull request merged**